### ISSUE TITLE
Iosevka Customizer website feature request: Ability to generate & download the font based on the variants chosen in ZIP archive.

### ISSUE THREAD

**Issue by ToddIndry37** [be5invis/Iosevka#2204]
Hi! I think customizing the Iosevka with different style variants is really cool & fun, but there is missing something. Maybe you can add ability to generate the font variants that user chosen & download the customized Iosevka into ZIP archives. This request was inspired from Commit Mono's customize webpage that they could download their own customized Commit Mono without generating them manually. I mean, not all people could generate their customized Iosevka without setting up NPM & the whole stuff to get started.

![screenshot-1708147098602](https://github.com/be5invis/Iosevka/assets/88038185/058e246b-f848-4a43-826c-2841264a032b)

**Comment by be5invis** [comment]
If @lemzwerg provides a WASM version of _ttfautohint_ then I could probably make it...
However, building Iosevka without a cache is very slow. Your browser may freeze for hours for an ab-initio build.

**Comment by lemzwerg** [comment]
Well, I won't provide a WASM version by myself but I invite interested parties to work on that...

**Comment by ToddIndry37** [comment]
Oh, I see there. Maybe some folks probably could contribute their WASM version of ``ttfautohint`` to this though, hopefully.

**Comment by ToddIndry37** [comment]
Hey @be5invis, I think I found a thing for _probably_ for my suggestions. I think you're interested with [FontFreeze](https://github.com/mutsuntsai/fontfreeze). It's similar to the Iosevka Customizer & that Commit Mono's customize webpage, but it _can_ generate the font without compromising user's browser cache & automatically downloads the results.

**Comment by be5invis** [comment]
@ToddIndry37  Iosevka's variant selector system is VERY COMPLICATED and require a lot of fine programming of the GSUB (like, it will decompose all the letters with diacritics...), so conventional font "freezer" won't work.

**Comment by ToddIndry37** [comment]
Oh, I see… maybe you can grab those generate font function thing & probably little modify it I guess?

**Comment by AndydeCleyre** [comment]
FWIW I have a [git repo](https://github.com/AndydeCleyre/archbuilder_iosevka/) who's purpose is to ease having GitHub Actions build Iosevka for you, and can be used with the output of the customizer site.

**Comment by ToddIndry37** [comment]
> FWIW I have a [git repo](https://github.com/AndydeCleyre/archbuilder_iosevka/) who's purpose is to ease having GitHub Actions build Iosevka for you, and can be used with the output of the customizer site.

@AndydeCleyre Wait, is that _really_ possible to get the output for Iosevka Customizer generator without using `ttfautohint`?

**Comment by AndydeCleyre** [comment]
> @AndydeCleyre Wait, is that _really_ possible to get the output for Iosevka Customizer generator without using `ttfautohint`?

No, it uses ttfautohint, sorry I didn't understand you were trying to avoid it.

**Comment by alexeyten** [comment]
@ToddIndry37 
> is that really possible to get the output for Iosevka Customizer generator without using ttfautohint?

You could build `ttf-unhinted::IosevkaCustom`

**Comment by ToddIndry37** [comment]
### Are there any updates for this? Also, I have some ideas after looking back my suggestion:

1. First of all, I think using GitHub Actions for exporting custom Iosevka font kinda neat though (thanks to @alexeyten for that idea!). Maybe I'll suggest you add some buttons on Customizer webpage to make those process happened, and then notify to the user who requested the customized Iosevka typeface after those export process completed via e-mail or GitHub notifications.
2. Secondly, this probably little off topic to this issue, but since Iosevka typeface kinda big (nearly 10 MB of size after extracting from archive), I think you need to little decrease those font files because it kinda cost a lot of storage when installing the font. Just little complain there.
3. Lastly, add more font styles! Like serif variant & probably the non-monospace version of Iosevka! It'll be awesome & add it to Iosevka Customizer (if you plan to do that).

That's it for now. Let me know if there's something I missed for suggestion for this issue (or literally any Iosevka topic related).

**Comment by be5invis** [comment]
@ToddIndry37 Your goal 2 and 3 are conflicting... Iosevka is this bug because it is this _complex_.
For goal 4 there's already a live preview in the customizer. At the right half of the screen.
![image](https://github.com/be5invis/Iosevka/assets/240091/cf4a3246-6cb4-4313-839d-3c0962534061)
Not having a textarea is because, well, Harfbuzz (the thing under the hood that applies variant glyphs) dislikes appliying too many featurea at one time, so textareas will be broken...

**Comment by ToddIndry37** [comment]
Ah, I see there. But I'll see if my first & second goal could achieve in the future of Iosevka Customizer thing.

**Comment by abhiramakella** [comment]
I absolutely agree that this needs to happen. I am not a pro of manually doing the custom build. 😔

**Comment by ToddIndry37** [comment]
> I absolutely agree that this needs to happen. I am not a pro of manually doing the custom build.

Same here. I don't have capable PC to manually build my own version of Iosevka since that requires a lot of resources in order to build the font. And yeah, both Maple Mono and Commit Mono has these font customization through web browser and can be exported directly through there.

**Comment by Logo121** [comment]
FYI, regarding your ideas
1. There is an [official template](https://github.com/be5invis/iosevka-custom-build-demo) to do this now, so if computing resource is the only problem, you can use it.
2. One reason for the large size is the amount of characters this font supports. If you don't need most of them and can do a custom build (e.g. via the above method), you can [limit the character set](https://github.com/be5invis/Iosevka/blob/main/doc/custom-build.md#limiting-the-character-set).
3. (tbh i have no idea what this means)
4. There is a half-functional live preview now thanks to #3224.
